geomesa-hbase/geomesa-hbase-datastore/src/main/scala/org/locationtech/geomesa/hbase/coprocessor/GeoMesaCoprocessor.scala

```diff
@@ -0,0 +1,155 @@
+  override def stop(coprocessorEnvironment: CoprocessorEnvironment): Unit = {
+  }
+
+  override def getService: Service = this
+
+  def getResult(controller: RpcController,
+                request: GeoMesaProto.GeoMesaCoprocessorRequest,
+                done: RpcCallback[GeoMesaProto.GeoMesaCoprocessorResponse]): Unit = {
+    val options: Map[String, String] = deserializeOptions(request.getOptions.toByteArray)
+    val aggregator = getAggregator(options)
+
+    val scanList: List[Scan] = getScanFromOptions(options)
+    val filterList: FilterList = getFilterListFromOptions(options)
+
+    val sft = SimpleFeatureTypes.createType("input", options(SFT_OPT))
+    val serializer = new KryoFeatureSerializer(sft, SerializationOptions.withoutId)
+
+    val response: GeoMesaCoprocessorResponse = try {
+
+      scanList.foreach(scan => {
+        scan.setFilter(filterList)
+        // TODO: Explore use of MultiRangeFilter
+        val scanner = env.getRegion.getScanner(scan)
+
+        try {
+          val results = new java.util.ArrayList[Cell]
+
+          def processResults() = {
+            for (cell <- results) {
+              // TODO: GEOMESA-1868 Set the Feature ID in the GeoMesa Coprocessor
+              val sf = serializer.deserialize(cell.getValueArray, cell.getValueOffset, cell.getValueLength)
+              aggregator.aggregate(sf)
+            }
+            results.clear()
+          }
+
+          @tailrec
+          def readScanner(): Unit = {
+            if(scanner.next(results)) {
+              processResults()
+              readScanner()
+            } else {
+              processResults()
+            }
+          }
+
+          readScanner()
+        } finally {
+          CloseQuietly(scanner)
+        }
+      })
+
+      val result: Array[Byte] = aggregator.encodeResult()
+      GeoMesaCoprocessorResponse.newBuilder.setSf(ByteString.copyFrom(result)).build
+    } catch {
+      case ioe: IOException =>
+        ResponseConverter.setControllerException(controller, ioe)
+        null
+      case cnfe: ClassNotFoundException =>
+        throw cnfe
+      case dse: DeserializationException =>
+        throw dse
+    }
+
+    done.run(response)
+  }
+}
+
+object GeoMesaCoprocessor {
+  /**
+    * It gives the combined result of pairs received from different region servers value of a column for a given column family for the
+    * given range. In case qualifier is null, a min of all values for the given
+    * family is returned.
+    *
+    * @param table
+    * @return HashMap result;
+    * @throws Throwable
+    */
+  def execute(table: Table, options: Array[Byte]): List[ByteString] = {
+    val requestArg: GeoMesaCoprocessorRequest = GeoMesaCoprocessorRequest.newBuilder().setOptions(ByteString.copyFrom(options)).build()
+
+    val callBack: GeoMesaHBaseCallBack = new GeoMesaHBaseCallBack()
+
+    table.coprocessorService(classOf[GeoMesaCoprocessorService], null, null, new Call[GeoMesaCoprocessorService, ByteString]() {
```

Comment:
Is there any reason to not pull the new Call into variable so this is easier to read?